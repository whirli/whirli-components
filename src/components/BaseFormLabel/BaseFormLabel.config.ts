// @ts-ignore
import { PropType } from 'vue';
import { ComponentStyles, ComponentProps } from '../../@types/components';

// Local config
import config from '@whirli-local/components/BaseFormLabel/BaseFormLabel.config';

// Package config
import { PropKeys, PropValues } from './BaseFormLabel.constants';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  [PropKeys.NAME]: {
    required: true,
    type: String as PropType<string>,
  },
  [PropKeys.REQUIRED]: {
    required: false,
    type: Boolean as PropType<boolean>,
    default: false,
  },
  ...config.props,
};
