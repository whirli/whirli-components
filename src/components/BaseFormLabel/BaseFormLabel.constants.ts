export const PropKeys: Record<string, string> = {
  REQUIRED: 'required',
  NAME: 'name',
};

export const PropValues: Record<string, string[]> = {};
