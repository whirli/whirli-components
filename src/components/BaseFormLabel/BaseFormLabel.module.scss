@import '@whirli-components/styles/Breakpoints.module.scss';

// default styles
.label {}

.required:after {
  content: ' *';
  display: inline;
}

// breakpoint styles
$packageBreakpointStyles: ();
