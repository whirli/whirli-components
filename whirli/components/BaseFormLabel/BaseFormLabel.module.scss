@import '@whirli-components/components/BaseFormLabel/BaseFormLabel.module.scss';

// Standard styles

// Breakpoint styles
$localBreakpointStyles: ();

@include createBreakpointClasses(map-merge($packageBreakpointStyles, $localBreakpointStyles));