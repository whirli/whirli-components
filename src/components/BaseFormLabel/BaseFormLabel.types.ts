import { PropKeys, PropValues } from './BaseFormLabel.constants';
