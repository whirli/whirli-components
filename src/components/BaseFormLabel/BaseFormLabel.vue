<template>
  <label :for="props.name" :class="[classes, requiredClass]"><slot /></label>
</template>

<script setup lang="ts">
// Styles
// @ts-ignore
import styles from '@whirli-local/components/BaseFormLabel/BaseFormLabel.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormLabel.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
const { makeClasses } = useClasses();
const classes: string[] = [styles.label, ...makeClasses(ComponentStyles, props, styles)];
const requiredClass: string = props.required ? styles.required : '';
</script>
