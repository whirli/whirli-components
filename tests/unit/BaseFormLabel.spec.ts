import { shallowMount } from '@vue/test-utils';
import BaseFormLabel from '@/components/BaseFormLabel/BaseFormLabel.vue';

let wrapper: any;
const text = 'Hello World';
const name = 'hello-world';

describe('BaseFormLabel.vue', () => {
  beforeEach(() => {
    wrapper = shallowMount(BaseFormLabel, {
      slots: {
        default: text,
      },
      props: {
        name: name,
      },
    });
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays the correct text', () => {
    expect(wrapper.text()).toMatch(text);
  });

  it('has the correct tag', () => {
    expect(wrapper.element.tagName).toBe('LABEL');
  });

  it('has the correct for property', () => {
    expect(wrapper.element.getAttribute('for')).toBe(name);
  });
});
