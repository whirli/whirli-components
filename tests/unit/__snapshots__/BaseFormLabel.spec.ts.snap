// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseFormLabel.vue renders the component 1`] = `
<label
  class=""
  for="hello-world"
>
  
  Hello World
  
</label>
`;
